src/components/Comment/Comment.tsx

```diff
@@ -0,0 +1,155 @@
+import FavoriteIcon from "@mui/icons-material/Favorite";
+import FavoriteBorderOutlinedIcon from "@mui/icons-material/FavoriteBorderOutlined";
+import InsertCommentOutlinedIcon from "@mui/icons-material/InsertCommentOutlined";
+import { Avatar, Box, Button, Divider, Typography } from "@mui/material";
+import { Timestamp } from "firebase/firestore";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useParams } from "react-router-dom";
+import { useAddReply } from "../../services/mutations/useAddReply";
+import { useUpdateCommentLikes } from "../../services/mutations/useUpdateCommentLikes";
+import { Comment as CommentType, NewComment } from "../../types/documentTypes";
+import { calculateTimeDifference } from "../../utils/calculateTimeDifference";
+import CommentTextfield from "./CommentTextfield";
+
+interface CommentProps {
+  comment: CommentType;
+  reply?: boolean;
+}
+
+interface FormValues {
+  comment: string;
+}
+
+const Comment: React.FC<CommentProps> = ({ comment, reply = false }) => {
+  const { recipeId } = useParams<{ recipeId: string }>();
+  const [openReply, setOpenReply] = useState<boolean>(false);
+  const { control, handleSubmit, reset } = useForm<FormValues>({
+    defaultValues: {
+      comment: "",
+    },
+  });
+  const { isSubmitting, addReply } = useAddReply();
+  const { isLiking, updateCommentLikes } = useUpdateCommentLikes();
+  const { author, createdAt, text, likedBy = [], id: commentId } = comment;
+  //change after implementinf authentication
+  const userId = "UgmRjQ854xMU3ST8LiVv";
+
+  const onSubmit = (data: FormValues) => {
```

Comment:
Every time the component re-renders, functions like onSubmit and toggleOpenReply are recreated. Using useCallback could help with memoizing these functions, although in this case, the performance gains may be minimal.